content-tool/tool/src/java/org/sakaiproject/content/tool/ListItem.java

```diff
@@ -4475,6 +4475,18 @@ public String getServiceName()
 		// This is used when asking if the styles of the service should be used.
 		return ServerConfigurationService.getString("ui.service", "Sakai");
 	}
-	
+
+	/**
+	 *Parse the id of the file returns the original file name 
+ 	 */
+	public String getOriginalFileName() {
+		String idArray[] = this.id.split("/");
+		int idLen = idArray.length;
+		String originalFileName = "";
+		if(idArray != null && idArray.length>0) {
```

Comment:
If you're pulling out `idLen` variable then you should use it consistently throughout the code. There's also no need for the `null` check here as the code would have already thrown a NPE on line 4483 if idArray was `null`.